src/index.html

```diff
@@ -154,5 +156,133 @@ <h2 id="precip-val">0%</h2>
                 </div>
             </div>
         </div>
+        <div class="middle-segment">
```

Comment:
We can make use of section tags for separating the top and middle sections, it will increase the readability.
You can also explore on semantic tags